File: tests/class/distance.swift
public class Point {
    var x: Double
    var y: Double
    public init() {
        x = 0
        y = 0
    }
    public init(x _x: Double, y _y: Double) {
        x = _x
        y = _y
    }
    var isOrigin: Bool {
        get {
            return x == 0 && y == 0
        }
    }
}

public func distance(first: Point, second: Point) -> Double {
    let delta = Point(x: first.x - second.x, y: first.y - second.y)
    return (delta.x * delta.x + delta.y * delta.y).squareRoot()
}

File: tests/class/distance.js
export function distance$first$second$(first, second) {
  const _x = first.x - second.x;

  const _y = first.y - second.y;

  const point = new Point();
  point.x = _x;
  point.y = _y;
  const delta = point;
  return Math.sqrt(delta.x * delta.x + delta.y * delta.y);
}
export class Point {}